source/write-operations/replace-documents.txt

```diff
@@ -0,0 +1,212 @@
+from the :atlas:`Atlas sample datasets </sample-data>`. 
+
+To learn how to create a free MongoDB Atlas cluster and load the sample
+datasets, see the :ref:`<java-rs-getting-started>` tutorial.
+
+.. include:: /includes/reactor-note.rst
+
+Replace Operation
+-----------------
+
+You can perform a replace operation in MongoDB by using the ``replaceOne()``
+method. This method removes all fields except the ``_id`` field from the first
+document that matches the search criteria. It then inserts the fields and values
+you specify into the document.
+
+Required Parameters
+~~~~~~~~~~~~~~~~~~~
+
+The ``replaceOne()`` method requires the following parameters:
+
+- A **query filter** document, which determines which documents to replace. For
+  more information about query filters, see the 
+  :manual:`Query Filter Documents section </core/document/#query-filter-documents>` in
+  the MongoDB Server manual. 
+
+- A **replace** document, which specifies the fields and values to insert in the new 
+  document.
+
+Replace One
+-----------
+
+To replace a single document in a MongoDB collection, call the ``replaceOne()`` method
+and pass the document you want to add. Then, pass the ``replaceOne()`` method to the static
+``Mono.from()`` method from the ``Mono Publisher``. The ``Mono Publisher`` is a
+``Publisher`` implementation from the Project Reactor library. In {+java-rs+}, you must use
+``Publisher`` implementations to control how data is transmitted in your
+application. To learn more about ``Mono``, see `Mono <https://projectreactor.io/docs/core/release/reference/#mono>`__ in the Project
+Reactor documentation.
+
+The following example uses the ``replaceOne()`` method to replace the fields and values of a
+document with a ``name`` field value of ``"Pizza Town"``. The ``replaceOne()``
+method replaces the original document with a document that has a ``name`` field
+value of ``"Mongo's Pizza"``.
+
+.. code-block:: java
+   :copyable: true
+
+    Mono.from(restaurants.replaceOne(
+      eq("name", "Pizza Town"),
+      new Document().append("name", "Mongo's Pizza")
+      .append("cuisine", "Pizza"))).block();
+
+Modify Replace Behavior
+-----------------------
+
+The ``ReplaceOptions`` class contains methods that modify
+the behavior of the ``replaceOne()`` method. To use the ``ReplaceOptions``
+class, construct a new instance of the class, then call one or more of its methods
+to modify the replace operation. You can chain these method calls together.
+To modify the behavior of the replace operation, pass the class instance and
+chained method calls as the second argument to the
+``replaceOne()`` method.
+
+You can use the following methods in the ``ReplaceOptions`` class
+to modify the ``replaceOne()`` method. All methods are optional.
+
+.. list-table::
+   :widths: 30 70
+   :header-rows: 1
+
+   * - Method
+     - Description
+
+   * - ``bypassDocumentValidation(Boolean bypassDocumentValidation)``
```

Comment:
S: I think you should present these methods without including the parameter type. Instead, you can describe the parameter in the description

```suggestion
   * - ``bypassDocumentValidation()``
```

Reply:
hey, the reason I added parameter type was because in all the other tables in the other write pages (insert, update, etc), we have the parameter type, so keeping consistency; it was also suggested as a edit in the past when I had not included the parameter type in the method. do you think that keeping it in the method makes it unclear?

Reply:
Oh, I didnt know that was a convention you had started in this docs set. I think you should remain consistent, meaning that you can disregard feedback related to this point